app/models/session.rb

```diff
@@ -131,18 +131,21 @@ def create_patient_sessions!
         required_programmes.all? { |programme| patient.vaccinated?(programme) }
       end
 
-    # First we remove patients from any other upcoming sessions.
+    # Mark existing patient sessions for transfer
     unvaccinated_patients.each do |patient|
-      sessions_other_than_self = patient.upcoming_sessions.reject { _1 == self }
-      next if sessions_other_than_self.empty?
+      other_sessions = patient.upcoming_sessions.where.not(id:)
```

Comment:
I think this'll cause a query to the database to run, which is why is originally gone with `reject`.